evalml/tests/automl_tests/test_iterative_algorithm.py

```diff
@@ -247,3 +246,41 @@ class MockTextClassificationPipeline(BinaryClassificationPipeline):
     assert pipeline.parameters['Text Featurization Component'] == expected_params
     assert isinstance(pipeline[0], TextFeaturizer)
     assert pipeline[0]._all_text_columns == ['text_col_1', 'text_col_2']
+
+
+@pytest.mark.parametrize("parameter", [1, "hello", 1.3, -1.0006, [1, 3, 4], (2, 3, 4)])
+def test_iterative_algorithm_pipeline_params(parameter, dummy_binary_pipeline_classes):
```

Comment:
Can you run more than one batch in this test and `test_iterative_algorithm_pipeline_params_skopt` ? Can be done in a separate test if you prefer that but I think we should cover that case!